plc4j/protocols/driver-bases/base/src/main/java/org/apache/plc4x/java/base/connection/AbstractPlcConnection.java

```diff
@@ -57,6 +67,42 @@ public boolean canSubscribe() {
         return false;
     }
 
+    /**
+     * The default implementation uses the {@link #ping(int)} method.
+     * Drivers that want to implement a more specific version have to overide it.
+     */
+    @Override
+    public boolean isConnected() {
+        return ping(PING_TIMEOUT_MS);
+    }
+
+    /**
+     * Method that sends a test-request or ping to the PLC to check if the PLC is still there.
+     * In most cases this method should only be used from the {@link #isConnected()} method.
+     * This method can only be used if ghe {@link #getInetSocketAddress()} returns a Socket Adress.
+     * Otherwise it throws a {@link NotImplementedException} to inform the user about that.
+     */
+    protected boolean ping(int timeout) {
+        Optional<InetSocketAddress> address = getInetSocketAddress();
```

Comment:
You could write this in one line with orElseThrow()

Reply:
done.